com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/AbortFailedProcessCommandFactory.java

```diff
@@ -50,43 +56,34 @@ public Object execute(CommandContext commandContext) {
 
         private void abortProcessIfRequested(CommandContext commandContext) {
             String processInstanceId = findProcessInstanceId(commandContext);
-            if (processInstanceId == null) {
-                return;
+            if (processInstanceId != null) {
+                abortProcessIfRequested(commandContext, processInstanceId);
             }
-            abortProcessIfRequested(commandContext, processInstanceId);
         }
 
         private String findProcessInstanceId(CommandContext commandContext) {
             JobEntity job = CommandContextUtil.getJobService(commandContext)
                                               .findJobById(jobId);
-            if (job == null) {
-                return null;
-            }
-            return job.getProcessInstanceId();
+            return job != null ? job.getProcessInstanceId() : null;
         }
 
         private void abortProcessIfRequested(CommandContext commandContext, String processInstanceId) {
             String correlationId = HistoryUtil.getVariableValue(commandContext, processInstanceId, Constants.VAR_CORRELATION_ID);
-            if (!processInstanceId.equals(correlationId)) {
-                return;
-            }
-            Boolean shouldAbortOnError = HistoryUtil.getVariableValue(commandContext, processInstanceId, Constants.PARAM_ABORT_ON_ERROR);
-            if (BooleanUtils.toBoolean(shouldAbortOnError)) {
-                abortProcess(commandContext, processInstanceId);
+            if (shouldAbort(commandContext, processInstanceId) || shouldAbort(commandContext, correlationId)) {
```

Comment:
Why the two checks?
If you end up here, the process with the correlationId(the super parent) will have the variable (ABORT_ON_ERROR). So, isn't only the second check enough?